src/dotnet/commands/dotnet-build/ProjectGraphCollector.cs

```diff
@@ -44,7 +44,7 @@ private ProjectGraphNode TraverseProject(ProjectDescription project, IDictionary
                     LibraryDescription libraryDescription;
                     if (lookup.TryGetValue(dependency.Name, out libraryDescription))
                     {
-                        if (libraryDescription.Identity.Type.Equals(LibraryType.Project))
+                        if (libraryDescription.Resolved && libraryDescription.Identity.Type.Equals(LibraryType.Project))
```

Comment:
And, an unresolved project library will not have any dependencies (since we couldn't find the project file) so even the else clause will no-op.